Pod/Classes/VOKCoreDataManager.m

```diff
@@ -319,46 +296,20 @@ - (NSArray *)importArray:(NSArray *)inputArray forClass:(Class)objectClass withC
         } else {
             returnObject = [self managedObjectOfClass:objectClass inContext:contextOrNil];
             [mapper setInformationFromDictionary:inputDict forManagedObject:returnObject];
+            if (inputKey) {
+                [existingObjectArray addObject:returnObject];
+                [existingUniqueKeys addObject:inputKey];
+            }
         }
         
-        [returnArray addObject:returnObject];
+        if (![returnArray containsObject:returnObject]) {
+            [returnArray addObject:returnObject];
+        }
```

Comment:
For any item to be imported, we do the importing above.  If the item existed before this batch-import, it gets updated.  If we already created the item during this batch-import, it gets updated (the case we care about).  If the item doesn't exist, we create it.

The `if (inputKey) …` bit above adds newly-created items to the array of existing items and its unique key to the existing unique keys (when there's a unique key), so that if we see another item to import with the same unique key, we update the one we created rather than creating a new one.

Here, we're making sure that the thing we just created/updated is in the return array—we add it only if it isn't already in the array.